Filter appointments with code

fawkesley opened the pull request:
This is introducing some filters like:
- before 10am
- with a female GP
- face to face

and using them to build the template context, rather than indexing straight
into the appointments array.

Commit:
Add `filterFaceToFace` function

And use this, rather than indexing directly into the right place in the
appointments.json file.

Commit:
Add `filterFemaleGP` function and use

Commit:
Add `filterBefore10` function

Commit:
Remove TODOs, we've resolved these particular ones

Commit:
Use `find_matching_appointment` wrapper

Instead of finding appointments with `filterFunctions(appointments)[0]`, now do
it like `find_matching_appointment([filters...])`

paulfurley commented:
Apologies for the inconsistency with camelCase vs snake_case - happy to resolve as another commit but don't want to rewrite the history just now

Commit:
Don't do `filter` inside the filter functions

Make the filter functions simply return boolean for a single
appointment, and do the `filter` call in the wrapper function.
